syntax ="proto3"; //声明protobuf的版本号

package fixbug; //声明代码所在的包（namespace）

//定义该选项，表示生成service服务类和rpc方法描述
option cc_generic_services=true;



message ResultCode
{
    int32 errcode=1;
    bytes errmsg=2;
}



message LoginRequest
{
    string name=1;
    string pwd=2;
}

message LoginResponse
{
    ResultCode result=1;
    bool success=2;//登录结果，是否成功
}

message User
{
    bytes name=1;
    int32 age=2;
    enum Sex
    {
        MAN=0;
        WOMAN=1;
    }
    Sex sex=3;
}

message FriendListRequest
{
    uint32 userid=1;
}

message FriendListResponse
{
    ResultCode result=1;
    repeated User friend_list=2;
}

service UserServiceRPC
{
    rpc Login(LoginRequest) returns(LoginResponse);
}

service FriendListServiceRPC
{
    rpc GetFriendList(FriendListRequest) returns(FriendListResponse);
}